### PR TITLE
Lesson 1 has passed

### PR DESCRIPTION
**Pull request by leobts**
All tasks have done from this lesson 1.
Content of tasks:
1. Computing
2. Using of math functions
3. Logics

### CODE REVIEW

File: src/lesson1/task1/Simple.kt [hunk 1]
```diff
@@ -60,31 +64,48 @@ fun seconds(hours: Int, minutes: Int, seconds: Int): Int = TODO()
...
 
 /**
  * Тривиальная
  *
  * Пользователь задает угол в градусах, минутах и секундах (например, 36 градусов 14 минут 35 секунд).
  * Вывести значение того же угла в радианах (например, 0.63256).
  */
-fun angleInRadian(grad: Int, min: Int, sec: Int): Double = TODO()
+fun angleInRadian(grad: Int, min: Int, sec: Int): Double {
+    val x = min/60
```
Comment (mglukhikh) [inline src/lesson1/task1/Simple.kt]: Int / Int = Int in Kotlin (and many other languages like C or Java). So e.g. if `min` is 30 your `x` is 30 / 60 = 0. To get Double result here, you should replace 60 with 60.0

File: src/lesson1/task1/Simple.kt [hunk 2]
```diff
@@ -102,12 +127,21 @@ fun travelMinutes(hoursDepart: Int, minutesDepart: Int, hoursArrive: Int, minute
  * Сколько денег будет на счету через 3 года (с учётом сложных процентов)?
  * Например, 100 рублей под 10% годовых превратятся в 133.1 рубля
  */
-fun accountInThreeYears(initial: Int, percent: Int): Double = TODO()
+fun accountInThreeYears(initial: Int, percent: Int): Double {
+    val z = (1+percent*0.01)*(1+percent*0.01)*(1+percent*0.01)
```
Comment (mglukhikh) [inline src/lesson1/task1/Simple.kt]: You can use Math.pow here

**Comment by mglukhikh** [comment]
Pay also attention to not working tests (see logs above).

File: src/lesson2/task1/IfElse.kt [hunk 3]
```diff
@@ -33,8 +33,11 @@ fun minBiRoot(a: Double, b: Double, c: Double): Double {
  * Мой возраст. Для заданного 0 < n < 200, рассматриваемого как возраст человека,
  * вернуть строку вида: «21 год», «32 года», «12 лет».
  */
-fun ageDescription(age: Int): String = TODO()
-
+fun ageDescription(age: Int): String {
+    val x = age % 10
+    return  if (x == 0) "$age лет" else if (x == 1)  "$age год"
```
Comment (mglukhikh) [inline src/lesson2/task1/IfElse.kt:38]: `return when (x)` looks nicer here

File: src/lesson2/task1/IfElse.kt [hunk 4]
```diff
@@ -79,7 +82,24 @@ fun rookOrBishopThreatens(kingX: Int, kingY: Int,
  * прямоугольным (вернуть 1) или тупоугольным (вернуть 2).
  * Если такой треугольник не существует, вернуть -1.
  */
-fun triangleKind(a: Double, b: Double, c: Double): Int = TODO()
+fun triangleKind(a: Double, b: Double, c: Double): Int {
+
+    val x = Math.acos((b * b + c * c - a * a) / 2.0 * b * c)
+    val y = Math.acos((a * a + c * c - b * b) / 2.0 * a * c)
+    val z = Math.acos((a * a + b * b - c * c) / 2.0 * a * b)
+    if (x > 90.0 || y > 90.0 || z > 90.0)
```
Comment (mglukhikh) [inline src/lesson2/task1/IfElse.kt:90]: Here you also can write code like
```
return when {
    x > 90.0 || y > 90.0 || z > 90.0 -> 2
    ...
}
```

**Comment by mglukhikh** [comment]
Mostly nice :)

**Comment by leobts** [comment]
Михаил, здравствуйте! Помогите. Запутался с заданием в 3 уроке (
* Найти количество цифр в заданном числе n.
* Например, число 1 содержит 1 цифру, 456 -- 3 цифры, 65536 -- 5 цифр.)

for надо использовать или нет?

fun check(n:Int) : Int {
    var sum = 1
    val m = Math.pow(10.0, sum-1.0)

    if (n<10) 1 else if (n>=10)
    for (????) {
        if (n / m > 0) continue
        sum ++
        if (n / m <0) break }
    return sum
}

**Comment by mglukhikh** [comment]
Sorry, test system is under refactoring yet.

File: src/lesson3/task1/Loop.kt [hunk 5]
```diff
@@ -114,7 +137,12 @@ fun squareBetweenExists(m: Int, n: Int): Boolean = TODO()
  * sin(x) = x - x^3 / 3! + x^5 / 5! - x^7 / 7! + ...
  * Нужную точность считать достигнутой, если очередной член ряда меньше eps по модулю
  */
-fun sin(x: Double, eps: Double): Double = TODO()
+fun sin(x: Double, eps: Double)
```
Comment (mglukhikh) [inline src/lesson3/task1/Loop.kt]: Looks awfully :(. Use loop here please. Also, please try to check the case when `x = 20 * PI` or even `x = 100 * PI`.

File: src/lesson3/task1/Loop.kt [hunk 6]
```diff
@@ -57,15 +57,38 @@ fun digitCountInNumber(n: Int, m: Int): Int =
...
  *
  * Найти число Фибоначчи из ряда 1, 1, 2, 3, 5, 8, 13, 21, ... с номером n.
  * Ряд Фибоначчи определён следующим образом: fib(1) = 1, fib(2) = 1, fib(n+2) = fib(n) + fib(n+1)
  */
-fun fib(n: Int): Int = TODO()
+fun fib(n: Int): Int {
+        var i = 1
+        var a = 0
+        var b = 1
+        while (i<=n) {
```
Comment (mglukhikh) [inline src/lesson3/task1/Loop.kt]: `for (i in 1..n)` is better here

File: src/lesson3/task1/Loop.kt [hunk 7]
```diff
@@ -57,15 +57,38 @@ fun digitCountInNumber(n: Int, m: Int): Int =
  * Найти количество цифр в заданном числе n.
  * Например, число 1 содержит 1 цифру, 456 -- 3 цифры, 65536 -- 5 цифр.
  */
-fun digitNumber(n: Int): Int = TODO()
+fun digitNumber(n: Int): Int {
+        var i = 1
+        var number = n
+        while (number > 0) {
+            if (number / 10 > 0) {
```
Comment (mglukhikh) [inline src/lesson3/task1/Loop.kt:64]: You can easily avoid this check

Comment (leobts) [reply src/lesson3/task1/Loop.kt:64]: How?

Comment (mglukhikh) [reply src/lesson3/task1/Loop.kt:64]: Think about starting value of `i`

File: src/lesson4/task1/List.kt [hunk 8]
```diff
@@ -148,7 +153,14 @@ fun polynom(p: List<Double>, x: Double): Double = TODO()
  * Например: 1, 2, 3, 4 -> 1, 3, 6, 10.
  * Пустой список не следует изменять. Вернуть изменённый список.
  */
-fun accumulate(list: MutableList<Double>): MutableList<Double> = TODO()
+fun accumulate(list: MutableList<Double>) :List<Double> {
+    if (list.isEmpty()) return list
+    for (i in 1..list.size-1) {
+        list[i] =  list[i] + list[i-1]
```
Comment (mglukhikh) [inline src/lesson4/task1/List.kt:159]: += can be used here

File: src/lesson4/task1/List.kt [hunk 9]
```diff
@@ -214,7 +226,26 @@ fun decimalFromString(str: String, base: Int): Int = TODO()
...
+fun roman(n: Int): String {
+    val numbers = intArrayOf(1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000, 5000)
+    val letters = arrayOf<String>("I", "IV", "V", "IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M", "MMMMM")
+    var romanValue = ""
+    var N = n
+    while (N > 0)
+    {
+        for (i in 0..numbers.size - 1)
+        {
+            if (N < numbers[i])
```
Comment (mglukhikh) [inline src/lesson4/task1/List.kt:283]: Very suspicious comparison. Please run local tests to check this function correctness.

**Pull request closed without merging**